### PR TITLE
chore: Rebase on upstream

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Rebases #6 on upstream RETH to unblock iteration.

### CODE REVIEW

File: crates/revm/src/executor.rs [hunk 1]
```diff
@@ -227,46 +232,177 @@ where
...
+                    Receipt {
+                        tx_type: transaction.tx_type(),
+                        // Success flag was added in `EIP-658: Embedding transaction status code in
+                        // receipts`.
+                        success: result.is_success(),
+                        cumulative_gas_used,
+                        // convert to reth log
+                        logs: result.into_logs().into_iter().map(into_reth_log).collect(),
+                        #[cfg(feature = "optimism")]
+                        deposit_nonce: None,
```
Comment (merklefruit) [inline crates/revm/src/executor.rs:402]: this is unnecessary as well

File: crates/revm/src/optimism.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+pub struct L1BlockInfo {
+    l1_base_fee: U256,
+    l1_fee_overhead: U256,
+    l1_fee_scalar: U256,
+}
+
+impl TryFrom<&Block> for L1BlockInfo {
+    type Error = executor::BlockExecutionError;
+
+    fn try_from(block: &Block) -> Result<Self, Self::Error> {
```
Comment (merklefruit) [inline crates/revm/src/optimism.rs:33]: good choice on TryFrom trait :)

**Review by merklefruit — APPROVED** [review]
LGTM

**Comment by clabby** [comment]
Moved to #8

**Pull request closed without merging**